kie-server-parent/kie-server-services/kie-server-services-jbpm-queries/src/main/java/org/kie/server/services/jbpm/queries/util/QueryStrategy.java

```diff
@@ -0,0 +1,33 @@
+/*
+ * Copyright 2017 Red Hat, Inc. and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+*/
+
+package org.kie.server.services.jbpm.queries.util;
+
+import java.util.Map;
+
+import org.jbpm.services.api.query.model.QueryParam;
+
+/**
+ * Provides DB specific query and column-mappings for queries.
+ * 
+ * @author <a href="mailto:duncan.doyle@redhat.com">Duncan Doyle</a>
+ */
+public interface QueryStrategy {
```

Comment:
I don't think we need these strategies, dasgbulder dataset should do that for us - at least this is what we rely on in workbench where we use same data sets across multiple dbs. Dashbuilder does retrieve that information based on db metadata so we should only provide type of fields that are known to dash builder 


or do I missing something big?